### ISSUE TITLE
Caproto incompatibility with Python 3.11

### ISSUE THREAD

**Issue by Hugzlb** [caproto/caproto#807]
Greetings everyone,

We tried to use caproto with Python 3.11 but unfortunately, the same error showed up every single time on different machines as soon as caproto's libraries were called. 

For Instance, the following small code:
```
from caproto.sync.client import read

PV_test = read('compteurCAPROTO')
print('compteurCAPROTO:',PV_test.data)
```

Always gave the following error:

```
Traceback (most recent call last):
  File "c:\Tests\caproto_issues.py", line 1, in <module>
    from caproto.sync.client import read
  File "C:\Users\hlebesco\AppData\Local\Programs\Python\Python311\Lib\site-packages\caproto\__init__.py", line 3, in <module>
    from ._broadcaster import *
  File "C:\Users\hlebesco\AppData\Local\Programs\Python\Python311\Lib\site-packages\caproto\_broadcaster.py", line 6, in <module>
    from ._commands import (Beacon, RepeaterConfirmResponse,
  File "C:\Users\hlebesco\AppData\Local\Programs\Python\Python311\Lib\site-packages\caproto\_commands.py", line 57, in <module>
    from ._status import eca_value_to_status, ensure_eca_value
  File "C:\Users\hlebesco\AppData\Local\Programs\Python\Python311\Lib\site-packages\caproto\_status.py", line 293, in <module>
    eca_value_to_status = {member.value.code_with_severity: member.value
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hlebesco\AppData\Local\Programs\Python\Python311\Lib\site-packages\caproto\_status.py", line 293, in <dictcomp>
    eca_value_to_status = {member.value.code_with_severity: member.value
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'tuple' object has no attribute 'code_with_severity'
```
_NB: The same issue appeared with others caproto libraries, such as caproto.threading.client._

In order to find a solution, we tried to run the same code with **Python 3.10** and it **worked properly**. Hence, we think that there might be an incompatibility between Caproto and Python 3.11, with both minimalist and complete caproto's installs.

We are well aware that, nowadays, caproto's support is limited but we wanted to let you know anyway. Maybe someone out there could be on the case.

Kind regards.

**Comment by danielballan** [comment]
Thanks!

**Comment by tacaswell** [comment]
I tested with the development branch of py311 (`Python 3.11.1+ (heads/3.11:c915f00eca, Jan  4 2023, 10:49:59) [GCC 12.2.0]`) and can not reproduce this so I very strongly suspect that this is a CPython bug. 

I do not have the time to bisect this to be sure, but I suspect that https://github.com/python/cpython/issues/100098 is the upstream bug and it is not worth changing to data classes to work around this.

I think for now the best bet is to drop to 3.11.0 and wait for 3.11.2.

I'm going to close this because it looks very much like an upstream bug that has already been fixed.  Even if we added "does not work with 3.11.1" to future releases, it would not help because the solver would just pick an old broken version (that we released before 3.11.1 and obviously could not predict the future).
